File: src/Java/Numbers On a Tree/NumberTree.java
import java.util.Scanner;

public class NumberTree {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String[] input = sc.nextLine().split(" ");
        if (input.length > 1) {
            int h = Integer.parseInt(input[0]);
            String path = input[1];
            int cur = 0; // initialized to index 0 at root
            for (char c : path.toCharArray()) {
                if (c == 'L') {
                    cur = cur * 2 + 1;
                } else {
                    cur = cur * 2 + 2;
                }
            }
            // Root value for perfect binary tree is 2 ^ (h + 1) - 1
            System.out.println((1 << h + 1) - 1 - cur);
        } else {
            int h = Integer.parseInt(input[0]);
            System.out.println((1 << h + 1) - 1);
        }
        sc.close();
    }
}

File: src/JavaScript/Numbers On a Tree/NumberTree.js
const rl = require('readline').createInterface(process.stdin, process.stdout);

rl.on('line', function(line) {
    let input = line.split(' ');
    if (input.length > 1) {
        const h = +input[0];
        const path = input[1];
        let current = 0;
        for (var i = 0; i < path.length; i++) {
            if (path.charAt(i) == 'L') {
                current = current * 2 + 1;
            } else {
                current = current * 2 + 2;
            }
        }
        console.log(h == 30 ? 2147483648 - 1 - current : (1 << h + 1) - 1 - current);
    } else {
        const h = +input[0];
        console.log(h == 30 ? 2147483648 - 1 : (1 << h + 1) - 1);
    }
    rl.close();
});